meps/mep-cross-references.md

```diff
@@ -0,0 +1,339 @@
+These URIs can be safely and easily parsed by any common URL parser. For example in Javascript:
+
+```js
+const url = new URL('project:target.md#my-ref');
+url.protocol; // "project:"
+url.pathname; // "target.md"
+url.hash; // "#my-ref"
+```
+
+For most internally linked references, we expect the inline syntax to be most commonly used, with the autolink and scheme to only be used when specific behavior is intended. The following links and references are supported:
+
+| Link Type                | Auto Link                  | Inline                    |
+| :----------------------- | :------------------------- | :------------------------ |
+| External URL             | `<https://example.com>`    | `[](https://example.com)` |
+| Local file download      | `<path:file.txt>`          | `[](file.txt)`            |
+| File download (explicit) | `<path:file.md>`           | `[](path:file.md)`        |
+| Project document         | `<project:file.md>`        | `[](file.md)`             |
+| Target in a document     | `<project:target.md#file>` | `[](file.md#target)`      |
+| Target in project        | `<project:#target>`        | `[](#target)`             |
+
+### Search Order and Specificity
+
+All references search the local document first[^specific_doc], then the local project in the order of the table of contents. A `xref_multiple` warning is raised if multiple matches are found.
+
+[^specific_doc]: With the exception of an explicit reference to a specific page, i.e. `[](./examples/my-doc.md#explicit-reference)`
+
+In large documentation sites, a referenced target can be present in multiple documents, in that case, the parser will emit a `xref_multiple` warning letting you know that there are multiple matches for the intended target.
+
+### Implicit Section Headers
+
+We suggest a configuration option to create anchor "slugs" for section headers, which stay close to the [GitHub implementation](https://github.com/Flet/github-slugger), which produces references that:
+
+- lower-case text
+- remove punctuation
+- replace spaces with `-`
+- enforce uniqueness via suffix enumeration `-1`
+
+For example, `## Links and Referencing` can be referenced as `[](#links-and-referencing)`.
+These are **implicit** references, and referring to them should raise an `xref_implicit` warning, which can optionally be suppressed by users.
+
+Implicit references are **not** available project wide, and are only accessible in the current document, as many documents follow similar structures (Abstract, Introduction, Methods, Summary). Adding two sections of the same name does not raise a duplicate identifier warnings (`xref_duplicate`), section identifiers are only unique to the document.
+
+### Paths
+
+- Links can be relative from the containing file.
+- Links can be absolute paths from the project root.
+- Absolute paths start with `/`.
+- The path separator is POSIX `/`.
+- The path must include the extension.
```

Comment:
Ah, I think this answers my previous question: no implicit resolution of `[](my-doc)` or `[](../supplementary/my-doc)` without extensions.

Reply:
Yeah, I think this is better and gets rid of ambiguity!!